content/kb/troubleshooting-volume-take-long-time-to-mount.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "Troubleshooting: Longhorn volumes take a long time to finish mounting"
+author: Phan Le
+draft: false
+date: 2021-02-26
+catelogies:
+  - "csi"
+---
+
+## Applicable versions
+All Longhorn versions.
+
+## Symptoms
+
+When starting workload pod which uses Longhorn volumes,
+observe in Longhorn UI that the Longhorn volumes are attached quickly
+but it takes a long time for the volumes to finish mounting and workload to be able to start.
+
+This issue only happens when Longhorn volumes that have many files/directories and the workload pod sets `fsGroup` in the `securityContext` similar to as below:
+```yaml
+spec:
+  securityContext:
+    runAsUser: 1000
+    runAsGroup: 3000
+    fsGroup: 2000
+```
+
+## Reason
+
+By default, when seeing `fsGroup` field, each time a volume is mounted, Kubernetes recursively `chown()` and `chmod()` all the files and directories inside the volume.
```

Comment:
By default, when seeing the `fsGroup` field, each time a volume is mounted, Kubernetes recursively calls `chown()` and `chmod()` on all the files and directories inside the volume.

Reply:
Thank you. Fixed